packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx

```diff
@@ -0,0 +1,401 @@
+
+  expect(screen.getByRole('tab')).not.toHaveAttribute('aria-label');
+});
+
+test('Renders with the toggleAriaLabel as an aria label on the tab', () => {
+  render(<OverflowTab toggleAriaLabel="Test aria label" />);
+
+  expect(screen.getByRole('tab')).toHaveAttribute('aria-label', 'Test aria label');
+});
+
+test('Renders the tab with aria-haspopup=menu', () => {
+  render(<OverflowTab />);
+
+  expect(screen.getByRole('tab')).toHaveAttribute('aria-haspopup', 'menu');
+});
+
+test('Renders the tab with aria-expanded set to false by default', () => {
+  render(<OverflowTab />);
+
+  expect(screen.getByRole('tab')).toHaveAttribute('aria-expanded', 'false');
+});
+
+test('Renders the tab with aria-expanded set to true when the menu is opened', () => {
+  render(<OverflowTab />);
+
+  const overflowTab = screen.getByRole('tab');
+
+  userEvent.click(overflowTab);
+
+  expect(overflowTab).toHaveAttribute('aria-expanded', 'true');
+});
+
+test('Passes Popper popperMatchesTriggerWidth set to false', () => {
+  render(<OverflowTab />);
+
+  // This assertion relies on the structure of the Popper mock to verify the correct props are being sent to Popper
+  expect(screen.getByText('Popper matches trigger width: false')).toBeVisible();
+});
+
+test('Passes Popper an appendTo value of the presentation element', () => {
+  render(<OverflowTab />);
+
+  userEvent.click(screen.getByRole('tab'));
+
+  // This assertion relies on the structure of the Popper mock to verify the correct props are being sent to Popper
+  expect(screen.getByText('Append to class name: pf-c-tabs__item pf-m-overflow')).toBeVisible();
+});
+
+test('Does not render an overflowing tab as a selected menu item by default', () => {
+  render(
+    <TabsContext.Provider value={{ ...tabsContextDefaultProps, localActiveKey: 0 }}>
+      <OverflowTab
+        overflowingTabs={[
+          { title: 'Tab one', eventKey: 1, tabContentRef: ('fakeRef' as unknown) as React.RefObject<any> }
+        ]}
+      />
+    </TabsContext.Provider>
+  );
+
+  userEvent.click(screen.getByRole('tab'));
+
+  // This assertion relies on the structure of the Popper mock to verify the correct props are being sent to Popper
+  expect(screen.queryByText('Selected: true')).not.toBeInTheDocument();
+});
+
+test('Renders an overflowing tab as a selected menu item when its key matches the localActiveKey provided via context', () => {
+  render(
+    <TabsContext.Provider value={{ ...tabsContextDefaultProps, localActiveKey: 1 }}>
+      <OverflowTab
+        overflowingTabs={[
+          { title: 'Tab one', eventKey: 1, tabContentRef: ('fakeRef' as unknown) as React.RefObject<any> }
+        ]}
+      />
+    </TabsContext.Provider>
+  );
+
+  userEvent.click(screen.getByRole('tab'));
+
+  // This assertion relies on the structure of the Popper mock to verify the correct props are being sent to Popper
+  expect(screen.getByText('Selected: true')).toBeVisible();
+});
+
+test('Matches snapshot', () => {
```

Comment:
Nit:

```suggestion
test('Matches snapshot when expanded', () => {
```

Just for clarifying the state of the component in the snapshot, but not a blocker.